### PR TITLE
Interactive Recording

### PR DESCRIPTION
**Pull request by dewski**
This PR brings interactive recording support to Cased Ruby (i.e. `rails console`, `irb`, etc.) as well as primitives necessary to create interactive approval flows.

### TODO

- [x] Update interactive prompts to match [cased/cli](https://github.com/cased/cli)

### CODE REVIEW

File: cased-ruby.gemspec [hunk 1]
```diff
@@ -33,7 +33,9 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'faraday', '~> 1.0'
   spec.add_dependency 'faraday_middleware', '~> 1.0'
   spec.add_dependency 'json', '~> 2'
+  spec.add_dependency 'jwt', '~> 2'
```
Comment (dewski) [inline cased-ruby.gemspec:36]: Need the JWT gem for generating and decoding JWT from Cased.

File: cased-ruby.gemspec [hunk 2]
```diff
@@ -33,7 +33,9 @@ Gem::Specification.new do |spec|
...
   spec.add_dependency 'net-http-persistent', '~> 3.0'
+  spec.add_dependency 'subprocess', '~> 1.5.0'
```
Comment (dewski) [inline cased-ruby.gemspec:38]: Used for recording CLI sessions.

File: lib/cased/cli/session.rb [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            @error = true
+            return false
+          end
+        end
+
+        response.success?
+      end
+
+      def cancel
+        response = Cased.clients.cli.post("#{api_url}/cancel", user_token: authentication.token)
```
Comment (dewski) [inline lib/cased/cli/session.rb:248]: Should probably expose a dictionary of URLs for each session in the REST API response so we don't need to build URLs by hand.

File: lib/cased/cli/log.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Cased
+  module CLI
+    module Log
```
Comment (tnm) [inline lib/cased/cli/log.rb:5]: Nice

File: lib/cased/cli/session.rb (hunk 3 above)
Comment (tnm) [reply lib/cased/cli/session.rb:248]: Yah agreed.

**Review by tnm — APPROVED** [review]
Exciting! 👍to merge and we’ll of course iterate after more use.

File: bin/cli [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+require 'cased-ruby'
+
+Cased.configure do |config|
+  config.guard_application_key = 'guard_application_1oFqltbMqSEtJQKRCAYQNrQoXsS'
+  config.guard_deny_if_unreachable = true
+end
+
+Cased::CLI::InteractiveSession.start
+
+puts 'Something destructive'
```
Comment (rikasun) [inline bin/cli:14]: 😂

**Pull request merged**